### ISSUE TITLE
how many epoch do you train when finetuning on IC15

### ISSUE THREAD

**Issue by kouxichao** [backtime92/CRAFT-Reimplementation#22]
some questions。
1.how many epoch do you train when finetuning on IC15
2.which epoch, the loss value began to decline significantly。

**Comment by kouxichao** [comment]
what's the hmean value on IC15 using pretrained model on synthtext.

**Comment by backtime92** [comment]
@kouxichao IC15 for 500 epochs,MLT for ~65 epochs. I forgot the IC15 hmean, maybe about 56-57%.

**Comment by kouxichao** [comment]
why is the hmean value of IC15 still very low , and go up and down between 0-0.3, after traing on IC15 for 200 epoch?  is that normal?

**Comment by backtime92** [comment]
@kouxichao Abnormal, it should be about ~80% for 200 epoch.

**Comment by kouxichao** [comment]
你使用IC15进行训练的时候，，，hmean怎么变化的？ 刚开始是0.0几，0.1几的情况吗?  我训练的时候为什么hmean一直在震荡，有时候0.1，有时候0.2。

**Comment by backtime92** [comment]
@kouxichao could you share you hmean?

**Comment by kouxichao** [comment]
this  is the loss figure generated in training process：
![loss_plot_training](https://user-images.githubusercontent.com/18493402/66899322-57475d00-f02d-11e9-9bb0-4edbd4ea6177.png)

the hmean value i didn't save.  screenshots of hmean on IC15:
![image](https://user-images.githubusercontent.com/18493402/66899598-d6d52c00-f02d-11e9-9172-27957436fee8.png)
![image](https://user-images.githubusercontent.com/18493402/66899621-e3f21b00-f02d-11e9-8260-2f3be38fc5b5.png)
![image](https://user-images.githubusercontent.com/18493402/66899660-f2403700-f02d-11e9-810e-38c36cc3b1af.png)
![image](https://user-images.githubusercontent.com/18493402/66899731-06843400-f02e-11e9-888a-d3aa07e691ca.png)
![image](https://user-images.githubusercontent.com/18493402/66899789-21ef3f00-f02e-11e9-9be9-ce0f96f6bbf8.png)
![image](https://user-images.githubusercontent.com/18493402/66899835-35020f00-f02e-11e9-94e5-b5f0fcdfc715.png)
![image](https://user-images.githubusercontent.com/18493402/66899853-3d5a4a00-f02e-11e9-8b0b-9f3a47b9ba4f.png)
![image](https://user-images.githubusercontent.com/18493402/66899875-45b28500-f02e-11e9-8dae-551b14137927.png)
![image](https://user-images.githubusercontent.com/18493402/66899893-4ea35680-f02e-11e9-97f9-6d14c64f61bf.png)
![image](https://user-images.githubusercontent.com/18493402/66899901-5662fb00-f02e-11e9-81ed-b779f6905354.png)

i retrained on IC15 for several times on IC15 for 100-200epoch， but the hmeans always are like above, i changed the batchsize to 1 for synthtext, and 5 for IC15, because the memory is not enough.
(i test the batchsize  2 for synthtext, and 5 for IC15 for 100 epoch , hmean is still like that. for memory reason , i only can train 100 epoch using original batchsize.)

**Comment by backtime92** [comment]
@kouxichao I will check it， and tomorrow give you the reason.

**Comment by kouxichao** [comment]
never mind, can you share your train info(the hmean and  loss  of every epoch)?, so that i can find something wrong happened much earlier in training. 
 or can you just simply describe the trend of hmean and loss change.  the hmean are always greater than the hmean(0.58) evaluted on pretrained model， or greater than some other value like 0.4?

And this pic is the input of net, is that normal? 
![image](https://user-images.githubusercontent.com/18493402/66995662-f9387980-f101-11e9-82bf-3c27c69adcf3.png)

**Comment by lianqing11** [comment]
@kouxichao Have you set the model to eval mode when generating the pseudo label code?

**Comment by kouxichao** [comment]
you mean this eval mode?
![image](https://user-images.githubusercontent.com/18493402/67055782-ff6b3c00-f17b-11e9-9f36-b53636a17d39.png)

**Comment by kouxichao** [comment]
@lianqing11 you mean this eval mode?
![image](https://user-images.githubusercontent.com/18493402/67055844-39d4d900-f17c-11e9-8ad4-28eb21d250ba.png)

**Comment by backtime92** [comment]
@lianqing11 thanks for @kouxichao ,just use the line 528 code, it will work.

**Comment by kouxichao** [comment]
sorry , line 528 code? i'm confused, can you tell me in detail?

**Comment by kouxichao** [comment]
@backtime92 sorry , line 528 code? i'm confused, can you tell me in detail?

**Comment by kouxichao** [comment]
![image](https://user-images.githubusercontent.com/18493402/67060438-5f1f1280-f18f-11e9-94be-4aae40020637.png)
you mean uncomment this line? i retrain with this uncommented, the hmean of epoch 0-17 is still like before, is that normal?

**Comment by backtime92** [comment]
@kouxichao data_loader.py file

**Comment by lianqing11** [comment]
@kouxichao Have you also set the model to train model --> model.train() after generating pseudo bounding box in every iteration, which may solve the problem.

**Comment by kouxichao** [comment]
you mean put net.train() in for loop? 
![image](https://user-images.githubusercontent.com/18493402/67060902-3b5ccc00-f191-11e9-8e74-7d54b4c7eed3.png)

**Comment by backtime92** [comment]
@kouxichao I am checking it, just for a moment.

**Comment by backtime92** [comment]
@kouxichao sorry， there is no leisure GPU， so i can not check it. Or can you train MLT, first?

**Comment by kouxichao** [comment]
thanks， i fixed it. i put net.train() at start and net.eval() at end of 'for' code block. it seems to work.

**Comment by kouxichao** [comment]
thanks， i fixed it. i put net.train() at start and net.eval() at end of 'for' code block. it seems to work.

**Comment by backtime92** [comment]
@kouxichao ok，sorry for the problems.

**Comment by kouxichao** [comment]
thanks for your great work and patience with the problem.

**Comment by backtime92** [comment]
for training epoch, the author trained the model for 16 images one GPU, so for weakly supervision, if you train the model for real image and syndata image with 10:2,  for IC15 you should train 500 epoch,  for MLT you should keep the iteartions * epoch = 25K.

**Comment by lerndeep** [comment]
> net.train()

could you please share this part of code?

**Comment by yakhyo** [comment]
Hi @kouxichao, I also having similar issue but even worse result on **143th epoch**:
```
Loading weights from checkpoint ./data/CRAFT-pytorch/real_weights/CRAFT_clr_143.pth
elapsed time : 86.13623404502869spytorch/test/img_85.jpgg
Calculated!{"precision": 0.0008183306055646482, "recall": 0.00048146364949446316, "hmean": 0.0006062443164595332, "AP": 0}
```
Could you share that which part of the code you have modified to fix the problem?
Thanks
